### ISSUE TITLE
[Issue] better error message when `azd up` is ran before python3.10-venv is installed

### ISSUE THREAD

**Issue by weikanglim** [Azure/azure-dev#59]
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
azd version 0.1.0-beta.1.1699704 (commit 4b105997b87ca4d36dab86d746afb19231b78a5e)

**Output from `az version`** 
{
  "azure-cli": "2.38.0",
  "azure-cli-core": "2.38.0",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "azure-devops": "0.25.0"
  }
}

**Describe the bug**
A cryptic error is printed when `azd restore` is ran before `python3.10-venv` is installed:
```bash
root@weilim-z4:~/repos/todo-python-mongo-wsl# azd restore
Error: requirements for project '/root/repos/todo-python-mongo-wsl/src/api' could not be installed: failed to install requirements for project '/root/repos/todo-python-mongo-wsl/src/api': exit status 2 (exit code: 2, stdout: , stderr: /bin/sh: 1: .: cannot open api_env/bin/activate: No such file
)
```

**To Reproduce**
1. Verify `venv` is not previously installed: `python3 -m venv --help` and you should get an error
2. Run `azd restore`
3. Examine error message:
```bash
root@weilim-z4:~/repos/todo-python-mongo-wsl# azd restore
Error: requirements for project '/root/repos/todo-python-mongo-wsl/src/api' could not be installed: failed to install requirements for project '/root/repos/todo-python-mongo-wsl/src/api': exit status 2 (exit code: 2, stdout: , stderr: /bin/sh: 1: .: cannot open api_env/bin/activate: No such file
)
```

**Expected behavior**
The error message should indicate something on how to solve the issue:
1. Install venv if not already
2. Remove the `api` folder
3. Rerun `azd restore`

Alternatively, if we can prevent the user from getting into this state.

**Environment**
Information on your environment:
    * Ubuntu 22.04 Linux

**Additional context**

**Comment by rajeshkamal5050** [comment]
Not a must-fix for Release 0.2. Keeping it in Backlog.

**Comment by weikanglim** [comment]
Closing this. We can create a new one if users hit this issue.
